pkg/util/util.go

```diff
@@ -108,3 +111,21 @@ func GetCanonicalTypeName(model interface{}) string {
 	}
 	return path + "." + t.Name()
 }
+
+// Provides a fast path for decoding YAML scalar node as a string
+// If the node's value can be simply returned directly, then it is. Otherwise,
+// the yaml.v3.Node.Decode slow path is taken
+func DecodeYAMLString(n *yaml.Node, s *string) error {
```

Comment:
What's the performance difference with this change? I'm curious if this might be better committed in the YAML library rather than here since there isn't any k8s specific logic.

/cc @apelisse

Reply:
+1 on not putting general-purpose yaml decoding in this lib

Reply:
Agree that it is not desirable long-term. I will prepare a PR for upstream. However, I do not see an alternative in the short term? It will likely take a while for the change to be merged and appear in a tagged release.

Reply:
@Jefftree The difference with this change is significant. Below are benchmark results for openapi v2 with this optimization removed:

```
goos: darwin
goarch: amd64
pkg: github.com/alexzielenski/parsebench
cpu: Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
BenchmarkFastConversion
BenchmarkFastConversion/json->swagger
BenchmarkFastConversion/json->swagger-8         	       2	 503133747 ns/op	95739432 B/op	 1381637 allocs/op
BenchmarkFastConversion/swagger->json
BenchmarkFastConversion/swagger->json-8         	       9	 122539288 ns/op	66432463 B/op	  274050 allocs/op
BenchmarkFastConversion/json->gnostic
BenchmarkFastConversion/json->gnostic-8         	       5	 225056310 ns/op	81462545 B/op	 1248402 allocs/op
BenchmarkFastConversion/gnostic->pb
BenchmarkFastConversion/gnostic->pb-8           	     100	  11495301 ns/op	 2899970 B/op	       1 allocs/op
BenchmarkFastConversion/pb->gnostic
BenchmarkFastConversion/pb->gnostic-8           	     100	  13719325 ns/op	 9480698 B/op	  123829 allocs/op
BenchmarkFastConversion/gnostic->yaml
BenchmarkFastConversion/gnostic->yaml-8         	      40	  30027913 ns/op	32855169 B/op	  264562 allocs/op
BenchmarkFastConversion/yaml->swagger
BenchmarkFastConversion/yaml->swagger-8         	      13	  92594648 ns/op	33392264 B/op	  675799 allocs/op
```

93.5ms in this run on my machine. Over a number of runs i saw a range of 80-100ms for yaml->swagger.

This is compared to ~60ms with the optimization enabled.